lib/src/contracts/contract_invoker.dart

```diff
@@ -0,0 +1,118 @@
+part of 'package:web3dart/contracts.dart';
+
+class ContractInvocation {
+  final Web3Client client;
+  final DeployedContract contract;
+  final ContractFunction function;
+  List _params;
+
+  ContractInvocation(this.client, this.contract, this.function);
+
+  ContractInvocation parameters([dynamic args]) {
+    if (args is Map) {
```

Comment:
This method overrides `_params`, which can be confusing when called twice. Let's add an
```suggestion
    assert(_params == null, 'Tried to call parameters multiple times');
    if (args is Map) {
```